### ISSUE TITLE
left join, bug when show data 

### ISSUE THREAD

**Issue by JuanIrigoyen** [aspnet/EntityFramework#6232]
### Steps to reproduce
1 - Create two entities with next structure.

 ```
public partial class Centros_trabajo_agrupacion
    {
        public Centros_trabajo_agrupacion()
        {
            Centros_trabajo = new HashSet<Centros_trabajo>();
        }
        public string Agrupacion { get; set; }
        public string Descripcion { get; set; }
        public string Observaciones { get; set; }

        public virtual ICollection<Centros_trabajo> Centros_trabajo { get; set; }
    }

public partial class Centros_trabajo
    {
        public string Codigo { get; set; }
        public string Descripcion { get; set; }
        public string Almacen { get; set; }
        public string Ubicacion { get; set; }
        public long Capacidad_diaria { get; set; }
        public string Tipo { get; set; }
        public string Observacion { get; set; }
        public string Agrupacion { get; set; }

        public virtual Centros_trabajo_agrupacion AgrupacionNavigation { get; set; }
    }
```

2 - Create context:

```
public partial class Maldivas_oranContext : DbContext
    {
        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
             optionsBuilder.UseSqlServer(@"Server=Sealand;Database=Maldivas_oran;User Id=test;Password=***");
        }
        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<Centros_trabajo>(entity =>
            {
                entity.HasKey(e => e.Codigo)
                    .HasName("PK_Centros");

                entity.ToTable("Centros_trabajo", "Produccion");
                entity.Property(e => e.Codigo).HasColumnType("nchar(6)");
                entity.Property(e => e.Agrupacion).HasColumnType("nchar(6)");
                entity.Property(e => e.Almacen).HasColumnType("nchar(3)");
                entity.Property(e => e.Capacidad_diaria).HasDefaultValueSql("288000000");
                entity.Property(e => e.Descripcion).HasMaxLength(200);
                entity.Property(e => e.Tipo).HasColumnType("nchar(3)");
                entity.Property(e => e.Ubicacion).HasColumnType("nchar(5)");

                entity.HasOne(d => d.AgrupacionNavigation)
                    .WithMany(p => p.Centros_trabajo)
                    .HasForeignKey(d => d.Agrupacion)
                    .HasConstraintName("FK_Centros_trabajo_Centros_trabajo_agrupacion");
            });

            modelBuilder.Entity<Centros_trabajo_agrupacion>(entity =>
            {
                entity.HasKey(e => e.Agrupacion)
                    .HasName("PK_Centros_trabajo_agrupacion");

                entity.ToTable("Centros_trabajo_agrupacion", "Produccion");
                entity.Property(e => e.Agrupacion).HasColumnType("nchar(6)");
                entity.Property(e => e.Descripcion).HasMaxLength(200);
                entity.Property(e => e.Observaciones).HasColumnType("ntext");
            });
        }

        public virtual DbSet<Centros_trabajo> Centros_trabajo { get; set; }
        public virtual DbSet<Centros_trabajo_agrupacion> Centros_trabajo_agrupacion { get; set; }

```
3. Add data and execute this query:

 ```
 using (Maldivas_oranContext context = new Maldivas_oranContext())
            {

                var qprevius =
                        from c in context.Centros_trabajo
                        join d in context.Centros_trabajo_agrupacion on c.Agrupacion equals d.Agrupacion into gj
                        where c.Agrupacion == "AG0S01" && c.Capacidad_diaria > 0
                        from dgp in gj.DefaultIfEmpty()

                        select new
                        {
                            Centro = c.Codigo,
                            Descripcion = dgp == null ? c.Descripcion : dgp.Descripcion,
                            Agrupacion = c.Agrupacion,
                            Activo = true,
                            Capacidad_diaria = c.Capacidad_diaria
                        };

                var result = qprevius.ToList();
              
                -- debug result
               }
}
```

### The issue

When I run a query with left outer join, this does not show the data correctly , 

1. The sample query in LinPad5 returns the values correctyl 
2. In Sql query runs correctly, but duplicate the ' Agrupacion ' field
3. When displaying data C # debug program, This step shows the wrong result.

### Further technical details

EF 7

Attach images.

![linqpad5](https://cloud.githubusercontent.com/assets/4987574/17359805/8f064e40-5968-11e6-8a13-d1f2d9361ff2.JPG)
![linqsql](https://cloud.githubusercontent.com/assets/4987574/17359808/939b64cc-5968-11e6-9ce7-a53da0316007.JPG)
![linqc](https://cloud.githubusercontent.com/assets/4987574/17359814/97c3f136-5968-11e6-9edb-e73b882b4f61.JPG)

**Comment by maumar** [comment]
Same root cause as https://github.com/aspnet/EntityFramework/issues/6318, reassigning to @smitpatel since he is working on it already

**Comment by smitpatel** [comment]
fixed via 5953d96f1bfcdd529f958de6bfb2305f09e34e08
